R/deployments.R

```diff
@@ -143,7 +143,10 @@ deploymentRecord <- function(name,
   c(standard, metadata)
 }
 
+dcfVersion <- "1"
```

Comment:
The meaning of one of the fields is changing, yes? What does this mean when we attempt to use those DCF files with an old `rsconnect` release?

Reply:
Ooh, I think I get what you mean now — we're adding a `version` field that will allow us to do more versioning in the future, but there's no intent to add any backward incompatible changes at this time. The current goal is to make it possible to tell if `applicationId` means `appId` or `contentId` in existing deployments to cloud; `contentId` is substantially easier to work with and more meaningful on the server side, but we don't want to break existing deployment records that stored `appId`. 

We considered renaming `applicationId` more generally (since rsconnect is not just about applications any more), but given that it's mostly an internal feature used by the IDE, we didn't think it was worth it.

Reply:
The situation that concerned me:

1. Deploy (new) using CRAN rsconnect (or any version prior to this change).
2. Re-deploy using this version of rsconnect (which upgrades the DCF file).
3. Re-deploy using CRAN rsconnect.

Is the deployment DCF value of `applicationId` constant or does it change during this re-deploy sequence?

Reply:
@aronatkins The value of `appId` in the DCF file would change when deploying using this version of rsconnect (from the application id to the content id), so that scenario would break at step 3.

Reply:
@hadley - You had mentioned trying to avoid this type of backwards incompatibility with the upcoming release.. has your opinion changed?

Reply:
The reasoning: There's enough changing in rsconnect that the upcoming release will cause issues for some folks despite our efforts to avoid that. Folks will probably downgrade to the previous CRAN release until we can patch. We wanted to avoid situations where the downgrade caused its own problems.

Reply:
I _think_ this is ok because of its relatively limited scope — my other proposed change would've broken every single piece of metadata that you'd saved, which seemed a bit extreme.